src/tools/fuzzing/heap-types.cpp

```diff
@@ -154,20 +154,27 @@ struct HeapTypeGeneratorImpl {
 
   HeapType::BasicHeapType generateBasicHeapType(Shareability share) {
     // Choose bottom types more rarely.
-    // TODO: string, exn, and cont types
+    // TODO: string and cont types
     if (rand.oneIn(16)) {
       HeapType ht =
         rand.pick(HeapType::noext, HeapType::nofunc, HeapType::none);
       return ht.getBasic(share);
     }
-    HeapType ht = rand.pick(HeapType::func,
-                            HeapType::ext,
-                            HeapType::any,
-                            HeapType::eq,
-                            HeapType::i31,
-                            HeapType::struct_,
-                            HeapType::array);
-    if (share == Unshared && features.hasSharedEverything() && rand.oneIn(2)) {
+
+    std::vector<HeapType> options{HeapType::func,
+                                  HeapType::ext,
+                                  HeapType::any,
+                                  HeapType::eq,
+                                  HeapType::i31,
+                                  HeapType::struct_,
+                                  HeapType::array};
```

Comment:
Given that we check for `features.hasExceptionHandling()` for `exn`, don't we also need to check for `features.hasGC()` or something for these?

Reply:
I think this entire file is for fuzzing of heap types, so GC is assumed everywhere. This is a separate file from the main fuzzer.